test/FunctionalTests/Tests/DataServices/UnitTests/ClientCSharpUnitTests/DataWebClientCSharp/BuildingRequestEventFunctionalTests.cs

```diff
@@ -962,9 +962,13 @@ private string RetrieveLastSegmentOfRequestUri(ReceivingResponseEventArgs arg)
             {
                 return arg.Descriptor.State.ToString();
             }
-            else
+            else if (arg.Descriptor != null)
             {
-                return (arg.ResponseMessage as HttpWebResponseMessage).Response.ResponseUri.Segments.Last();
+                return arg.Descriptor.State.ToString();
+            }
+            {
+                // this is the top-level batch request
+                return "$batch";

```

Comment:
What's with the code block here? It's not even an `else` block and it didn't exist before...